### PR TITLE
Rearchitect subscriptions

### PR DESCRIPTION
**Pull request by jellegerbrandy**

### CODE REVIEW

**Comment by jellegerbrandy** [comment]
This looks all pretty rational and sensible, but i do not see any signifcant gain in the number of subscriptions being sent to the subgraph. 

----

This is the number of subscriptions tht alchemy opens, measured by opening devtools, look at network tab, 
find the subgraph.daostack.io websocket, and check the id of the last subscription. 
Running this locally, agains tthe test setup in the docker-compose file.


| what | new  (this PR,  | old (dev, 72c3ca177493d9be21320d673a91f504ed3d871f)|
|-------|------|---|
| http://127.0.0.1:3000/ | 10 | 14 |
| navigate from main to http://127.0.0.1:3000/dao/0xe7a2c59e134ee81d4035ae6db2254f79308e334f | 36 | 35 |
| reload http://127.0.0.1:3000/dao/0xe7a2c59e134ee81d4035ae6db2254f79308e334f | 36 (sometimes 54?) | 30 or 54 |
| go from there to history http://127.0.0.1:3000/dao/0xe7a2c59e134ee81d4035ae6db2254f79308e334f/history/ | 48 | 58 |
| reload history: http://127.0.0.1:3000/dao/0xe7a2c59e134ee81d4035ae6db2254f79308e334f/history/ | 29 | 31 |

**Comment by dkent600** [comment]
@jellegerbrandy My recollection is that the number of subscriptions is meant to decrease in cases where render is called repeatedly in the absence of property changes, but not during the initial loading of pages.  The latter case might be addressed if we move some of the data loading up the component hierarchy, which I suspect Tibet hasn't done yet

**Comment by jellegerbrandy** [comment]
There is abug: go to (say) https://alchemy-staging-rinkeb-pr-1049.herokuapp.com/dao/0x72939947e7a1c4ac94bb840e3304b322237ad1a8/proposal/0x62140db91fd7fc19cd43ab9f024ac5719142f1298349d877389f2aeaa1c7ffd4, then click on "alchemy" in top right, you'll get a "Report feedback
TypeError: Cannot read property 'address' of null Error

**Comment by jellegerbrandy** [comment]
- on the preview app, I see the following error messages in the console, which we do not have in other prview apps: 
image

File: src/components/Daos/DaoCard.tsx [hunk 1]
```diff
@@ -3,86 +3,92 @@ import { DAO, IDAOState,
...
 };
 
-export default DaoCard;
+export default withSubscription({
+  wrappedComponent: DaoCard,
+  loadingComponent: null,
+  errorComponent: (props) => <div>{ props.error.message }</div>,
+
+  checkForUpdate: (oldProps, newProps) => {
+    // TODO: does dao.id work here or do we need to load the static state?
```
Comment (jellegerbrandy) [inline src/components/Daos/DaoCard.tsx]: should work :-)

**Comment by tibetsprague** [comment]
> * on the preview app, I see the following error messages in the console, which we do not have in other prview apps:
>   image

@jellegerbrandy this error already existed we just were not showing it before :) those token contracts don't exist...

**Comment by tibetsprague** [comment]
> There is abug: go to (say) https://alchemy-staging-rinkeb-pr-1049.herokuapp.com/dao/0x72939947e7a1c4ac94bb840e3304b322237ad1a8/proposal/0x62140db91fd7fc19cd43ab9f024ac5719142f1298349d877389f2aeaa1c7ffd4, then click on "alchemy" in top right, you'll get a "Report feedback
> TypeError: Cannot read property 'address' of null Error

fixed

**Comment by tibetsprague** [comment]
@jellegerbrandy looking on staging i see way way way more subscriptions being created than with the refactor. clicking around a couple times i very quickly get up to 300 subscriptions whereas on the new one i stay around 70

**Comment by tibetsprague** [comment]
try adding a few new proposals, i got up into the 1000s of subscriptions on staging

**Pull request merged**